sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/SnapshotImpl.java

```diff
@@ -125,6 +125,11 @@ public Boolean awaitCopyStartCompletion(Duration maxWaitTime) {
 
     @Override
     public Mono<Void> awaitCopyStartCompletionAsync() {
+        if (creationMethod() != DiskCreateOption.COPY_START) {
+            return Mono.error(logger.logThrowableAsError(new IllegalStateException(
+                String.format("Cannot call [awaitCopyStartCompletionAsync] on snapshot: %s, since creationMethod is [%s].",
```

Comment:
maybe something like this?

"awaitCopyStartCompletionAsync" cannot be called on snapshot when "creationMethod" is not "CopyStart"

Reply:
changed